### ISSUE TITLE
Custom backends with errors are flagged as READY

### ISSUE THREAD

**Issue by SlipknotTN** [NVIDIA/tensorrt-inference-server#285]
How to reproduce it:
- Use addsub example
- Change config.pbtxt setting wrong output names

e.g.

```
name: "add_sub_custom"
platform: "custom"
max_batch_size: 0
default_model_filename: "libaddsub.so"
input [
  {
    name: "INPUT0"
    data_type: TYPE_FP32
    dims: [ 16 ]
  },
  {
    name: "INPUT1"
    data_type: TYPE_FP32
    dims: [ 16 ]
  }
]
output [
  {
    name: "OUTPUT0_AAAAAA"
    data_type: TYPE_FP32
    dims: [ 16 ]
  },
  {
    name: "OUTPUT1"
    data_type: TYPE_FP32
    dims: [ 16 ]
  }
]
```

instead of
```
name: "add_sub_custom"
platform: "custom"
max_batch_size: 0
default_model_filename: "libaddsub.so"
input [
  {
    name: "INPUT0"
    data_type: TYPE_FP32
    dims: [ 16 ]
  },
  {
    name: "INPUT1"
    data_type: TYPE_FP32
    dims: [ 16 ]
  }
]
output [
  {
    name: "OUTPUT0"
    data_type: TYPE_FP32
    dims: [ 16 ]
  },
  {
    name: "OUTPUT1"
    data_type: TYPE_FP32
    dims: [ 16 ]
  }
]
```

You get this error in server log 
```
E0510 08:34:22.755094 1000 dynamic_batch_scheduler.cc:162] Initialization failed for dynamic-batch scheduler thread 0: initialize error for 'add_sub_custom': (6) model outputs must be named 'OUTPUT0' and 'OUTPUT1'
```

but model is flagged as READY (successfully loaded and MODEL_READY status).

If try to call the model, the client request hangs.
Is it an intended behaviour? I think this case, custom model should not be loaded.

**Comment by GuanLuo** [comment]
This is not the expected behavior. We are aware of the issue that the server is not notified with errors during custom backend initialization. After this is fixed, the model state should reflect the actual state of the custom backend.

**Comment by GuanLuo** [comment]
This issue should be fixed by #387, closing this issue.
